python/main.py

```diff
@@ -197,3 +197,105 @@ async def get_image(image_filename):
+def get_items_list():
+    # Connect to database and fetch all items
+    con = sqlite3.connect(db_path)
+    con.row_factory = sqlite3.Row
+    cur = con.cursor()
+    rows = cur.execute("""
+                    SELECT Drafts.draft_id AS id,
+                            Drafts.sort_index,
+                            Drafts.item_name AS name,
+                            Drafts.item_image_filename AS image,
+                            Categories.category_name AS category,
+                            Item_State.state AS state,
+                            Drafts.description AS description,
+                            Delivery.delivery_method AS delivery,
+                            Drafts.price AS price
+                    FROM Drafts
+                    LEFT JOIN Categories USING (category_id)
+                    LEFT JOIN Item_State USING (item_state_id)
+                    LEFT JOIN Delivery USING (delivery_method_id)
+                    """).fetchall()
+    con.close()
+
+    # Create a dictionary for each item
+    list = []
+    for row in rows:
+        columns = row.keys()
+        list.append({key: row[key] for key in columns})
+
+    # Return all items
+    return {"draft items": list}
+
+# POST request to post a single item to the drafts table in SQLite3 database
+@app.post('/drafts')
+async def add_draft(item_name: str = Form(...), image: UploadFile = File(...),
+category: str = Form(...), item_state_id: int = Form(...), description: str = Form(...),
+delivery_id: int = Form(...), price: int = Form(...)):
+    try:
+        image_binary = image.file.read()
+        db_image = open(f'draft_images/{image.filename}', "xb")
+        db_image.write(image_binary)
+        db_image.close()
+    except:
+        logger.info("New image could not be created:", sys.exc_info()[0])
+
+    con = sqlite3.connect(db_path)
+    cur = con.cursor()
+    # Use null for id value to allow persistent primary key to automatically generate
+    cur.execute("INSERT OR IGNORE INTO Categories VALUES (null, ?)", (category,))
+    cat_id = cur.execute("SELECT category_id FROM Categories WHERE category_name = ?", (category,)).fetchone()[0]
+    params = (item_name, image.filename, cat_id, item_state_id, description, delivery_id, price,)
+    cur.execute("""INSERT INTO Drafts (item_name, item_image_filename, category_id, item_state_id,
+     description, delivery_method_id, price)
+     VALUES (?, ?, ?, ?, ?, ?, ?)
+     """, params)
+    con.commit()
+    con.close()
+
+    # Return accepted item message
+    return {"message": f"item received in drafts: {item_name}, category: {category}, image_filename: {image.filename}"}
+
+# GET request to obtain the info a single item in the drafts table by inputting draft_id
+@app.get('/drafts/{draft_id}')
+def get_draft(draft_id: int):
+    con = sqlite3.connect(db_path)
+    con.row_factory = sqlite3.Row
+    cur = con.cursor()
+    # Use * to allow for schema changes (Not safe to add sensitive data to schema)
```

Comment:
Nit: this comment can be removed I think since this endpoint doesn't use `*` in the SQL commands ^^

Reply:
> Nit: The comment on line 268 can probably be removed at some point since this endpoint doesn't use `*` in the SQL commands ^^

thank you for pointing this out! I forgot to read the comments and remove those that are unnecessary :>